flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStorage.java

```diff
@@ -107,6 +104,12 @@ public boolean supportsHighlyAvailableStorage() {
 		return true;
 	}
 
+	@Override
+	public void initializeBaseLocations() throws IOException {
+		fileSystem.mkdirs(sharedStateDirectory);
+		fileSystem.mkdirs(taskOwnedStateDirectory);
+	}
+
 	@Override
 	public CheckpointStorageLocation initializeLocationForCheckpoint(long checkpointId) throws IOException {
 		checkArgument(checkpointId >= 0);
```

Comment:
Should we verify that the base locations have been initialized? Some kind of flag in `initializeBaseLocations()`?
The javadoc clearly states this contract, but it may help if we introduce a bug later.